### ISSUE TITLE
Incompatible with latest CHEFDK versions

### ISSUE THREAD

**Issue by calin-ciobanu** [rmoriz/knife-digital_ocean#89]
Hi, I am having trouble using the plugin with the latest versions of chefdk

```
OS: ubuntu 14.04
chefdk: 0.19.6 / 0.18.30
knife-digital_ocean: 2.5
```

when using knife I get:

```
knife digital_ocean image list
/opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:2288:in `raise_if_conflicts': Unable to activate resource_kit-0.1.5, because addressable-2.4.0 conflicts with addressable (~> 2.3.6) (Gem::ConflictError)
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1408:in `activate'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1442:in `block in activate_dependencies'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1428:in `each'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1428:in `activate_dependencies'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1410:in `activate'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1442:in `block in activate_dependencies'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1428:in `each'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1428:in `activate_dependencies'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/specification.rb:1410:in `activate'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems.rb:220:in `rescue in try_activate'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems.rb:213:in `try_activate'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:126:in `rescue in require'
    from /opt/chefdk/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:40:in `require'
    from /home/calin/.chefdk/gem/ruby/2.3.0/gems/knife-digital_ocean-2.5.0/lib/chef/knife/digital_ocean_size_list.rb:13:in `<top (required)>'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:100:in `load'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:100:in `block in load_commands'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:100:in `each'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:100:in `load_commands'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:110:in `load_command'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife/core/subcommand_loader.rb:124:in `command_class_from'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife.rb:153:in `subcommand_class_from'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/knife.rb:214:in `run'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/lib/chef/application/knife.rb:156:in `run'
    from /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.15.19/bin/knife:25:in `<top (required)>'
    from /usr/bin/knife:57:in `load'
    from /usr/bin/knife:57:in `<main>'
```

**Comment by calin-ciobanu** [comment]
after going through the dependencies I believe the issue is from : https://github.com/google/google-api-ruby-client/blob/master/google-api-client.gemspec#L25

**Comment by brianmd** [comment]
1+

**Comment by rmoriz** [comment]
the Digital Ocean adapter still requires `'addressable', '~> 2.3.6'` via https://github.com/digitalocean/resource_kit/blob/master/resource_kit.gemspec#L22

However chefdk already resolved to `2.4.0`: https://github.com/chef/chef-dk/blob/master/Gemfile.lock

Nothing much I can do here. Please ask `resource_kit` to bump the dependency.

**Comment by calin-ciobanu** [comment]
thanks!

**Comment by jtatum** [comment]
Looks like upstream fixed addressable, but now using chefdk 2.2.1, I get `Unable to activate droplet_kit-1.3.3, because faraday-0.13.1 conflicts with faraday (~> 0.9.1)`

**Comment by rmoriz** [comment]
It's probably time to sunset this project. Have a look at https://www.terraform.io/ and it's DigitalOcean and Chef-Provisioning (or use just the shell provisioning method to roll a custom setup). 

https://www.terraform.io/docs/providers/do/index.html
https://www.terraform.io/docs/provisioners/chef.html
https://www.terraform.io/docs/provisioners/local-exec.html
https://www.terraform.io/docs/provisioners/remote-exec.html
